drivers/include/periph/gpio.h

```diff
@@ -77,11 +77,33 @@ typedef unsigned int gpio_t;
 #define GPIO_PIN(x,y)       ((gpio_t)((x & 0) | y))
 #endif
 
+#ifndef GPIO_EXT_DEV_LOC
+/**
+ * @brief   Set device ID location in gpio_t when using extensions
+ */
+#define GPIO_EXT_DEV_LOC    (8*sizeof(gpio_t) - 8)
```

Comment:
Here, we will get a problem for `cpu/atmega_common/include/periph_cpu_common.h`. The following was added with merge of PR #10498:
https://github.com/RIOT-OS/RIOT/blob/2ebb187ca9b5413ef2ef5ab64e53d2e953d22482/cpu/atmega_common/include/periph_cpu_common.h#L37-L40

We should ask @lebrush why he changed the type of `gpio_t` with the PWM implementation. Either this definition has to removed from here or we also need to add a `GPIO_EXT_DEV_LOC` definition in `cpu/atmega_common/include/periph_cpu_common.h`

Reply:
Argh. Those changes should have been made in a separate PR. Also note the incompatible `GPIO_UNDEF` definition. This PR needs to be finalized and merged soon to reduce the accumulation of incompatible changes we need to undo.

Setting it back to at least `uint16_t` would be better, but we could get by with using the alternative encoding from `cpu/lpc1768/include/periph_cpu.h`, though this limits the user to 8 extension devices with 16 pins each. The `GPIO_UNDEF` definition needs to be removed.